pkg/kubectl/cmd/testdata/diff/test_diff.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+if command -v git &>/dev/null; then
+    git diff --no-index "${1}" "${2}" |grep containerPort -A 1 > "./git-diff.result"
+elif command -v diff &>/dev/null; then
+    diff "${1}" "${2}" |grep containerPort > "./diff.result"
+else
```

Comment:
I don't need both of these happen, if `git diff` exist it will use `git diff` and omit `diff`, I also update unit tests to make sure as long as one of those pass tests will pass, and this script is similar to `test_editor.sh`, I think this is fine.